cluster-essentials/test/clusterinit_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package test
+package test
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	cp "github.com/otiai10/copy"
+	"github.com/stretchr/testify/require"
+
+	clusteressentials "github.com/vmware-tanzu/tanzu-framework/cluster-essentials"
+	"github.com/vmware-tanzu/tanzu-framework/cluster-essentials/fakes"
+)
+
+var clusterConfig = &clusteressentials.ClusterEssentialsConfiguration{}
+var fake = &fakes.FakeClient{}
+
+func Test_InitClusterEssentials(t *testing.T) {
+	_ = os.MkdirAll("./tmp/package", os.ModePerm)
+	_ = cp.Copy("./testdata/package1", "./tmp/package")
+
+	t.Run("fails when unable to download cluster essential bundle ", func(t *testing.T) {
+		fake.DownloadClusterEssentialBundleReturns(errors.New("could not parse reference"))
+		clusterConfig.PkgClient = fake
+		err := clusterConfig.InitClusterEssentials()
+		require.Error(t, err)
+		require.ErrorContains(t, err, "unable to download cluster essential manifest")
+	})
+	t.Run("fails when unable to process downloaded cluster essential bundle ", func(t *testing.T) {
+		fake.DownloadClusterEssentialBundleReturns(nil)
+		var result1 []byte
+		fake.CarvelPackageProcessorReturns(result1, errors.New("error while running ytt"))
+		clusterConfig.PkgClient = fake
+		err := clusterConfig.InitClusterEssentials()
+		require.Error(t, err)
+		require.ErrorContains(t, err, "unable to process carvel package")
+	})
+	t.Run("fails when unable to process downloaded cluster essential bundle ", func(t *testing.T) {
```

Comment:
What's the difference between this test case and the previous one?

Reply:
No difference , copy paste error.